test/axios2.test.js

```diff
@@ -595,4 +596,393 @@ describe("Axios AuthHttpRequest class tests", function() {
+
+                let jwt = accessTokenPayload.jwt;
+
+                // Decode the JWT
+                let decodeResponse = await axios.post(`${BASE_URL}/jsondecode`, JSON.stringify({ jwt }), {
+                    headers: {
+                        Accept: "application/json",
+                        "Content-Type": "application/json"
+                    }
+                });
+
+                let decodedJWT = await decodeResponse.data;
+
+                // Verify the JWT claims
+                assertEqual(decodedJWT.sub, userId);
+                assertEqual(decodedJWT._jwtPName, undefined);
+                assertEqual(decodedJWT.iss, "http://0.0.0.0:8080");
+                assertEqual(decodedJWT.customClaim, "customValue");
+
+                // Update access token payload
+                await axios.post(
+                    `${BASE_URL}/update-jwt`,
+                    JSON.stringify({
+                        ...accessTokenPayload,
+                        customClaim: undefined,
+                        newClaim: "newValue"
+                    }),
+                    {
+                        headers: {
+                            Accept: "application/json",
+                            "Content-Type": "application/json"
+                        }
+                    }
+                );
+
+                // Get access token payload
+                accessTokenPayload = await supertokens.getAccessTokenPayloadSecurely();
+
+                // Verify new access token payload
+                assertNotEqual(accessTokenPayload.jwt, undefined);
+                assertEqual(accessTokenPayload.sub, userId);
+                assertEqual(accessTokenPayload._jwtPName, "jwt");
+                assertEqual(accessTokenPayload.iss, "http://0.0.0.0:8080");
+                assertEqual(accessTokenPayload.customClaim, undefined);
+                assertEqual(accessTokenPayload.newClaim, "newValue");
+
+                jwt = accessTokenPayload.jwt;
+
+                decodeResponse = await axios.post(`${BASE_URL}/jsondecode`, JSON.stringify({ jwt }), {
+                    headers: {
+                        Accept: "application/json",
+                        "Content-Type": "application/json"
+                    }
+                });
+
+                decodedJWT = await decodeResponse.data;
+
+                // Verify new JWT
+                assertEqual(decodedJWT.sub, userId);
+                assertEqual(decodedJWT._jwtPName, undefined);
+                assertEqual(decodedJWT.iss, "http://0.0.0.0:8080");
+                assertEqual(decodedJWT.customClaim, undefined);
+                assertEqual(decodedJWT.newClaim, "newValue");
+            });
+        } finally {
+            await browser.close();
+        }
+    });
+
+    it("Test that access token payload and JWT are valid after the property name changes and payload is updated after the session is created", async function() {
```

Comment:
Change to "after the session is refreshed"

Reply:
Removed the last few words, they didnt mean anything and were making the test name confusing